torch/distributed/_tensor/dispatch.py

```diff
@@ -198,11 +196,22 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
         # communicate the result to all ranks for some operators that return scalar value
         if output_sharding.output_spec is None:
             if op_call == aten.equal.default:
-                obj_list = [None for _ in range(dist.get_world_size())]
-                dist.all_gather_object(obj_list, local_results)  # type: ignore[possibly-undefined]
```

Comment:
Could you briefly explain the benefit of removing object collectives? Maybe consider updating the PR summary.

Reply:
1) `all_gather_object()` cannot be fully traced.
2) object collective has extra host-to-device mem copy and reportedly several SEVs were associated with it.